### PR TITLE
Replace hardcoded spacing values with tokens

### PR DESCRIPTION
**Pull request by aveline**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-react/issues/4726

### CODE REVIEW

File: UNRELEASED-v8.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ Use [the changelog guidelines](/documentation/Versioning%20and%20changelog.md) t
 - Removed the `z-index()` scss function ([#4753](https://github.com/Shopify/polaris-react/pull/4753))
 - Removed `nonDesignLangaugeCustomProperties` and `designLangaugeCustomProperties` ([#4770](https://github.com/Shopify/polaris-react/pull/4770))
 - Removed the `rem()` scss function and any references ([#4761](https://github.com/Shopify/polaris-react/pull/4761/))
+- Replaced hardcoded spacing values with spacing tokens ([#4775](https://github.com/Shopify/polaris-react/pull/4775))
```
Comment (kyledurand) [inline UNRELEASED-v8.md]: Not really a breaking change. What about nested under Code quality?

**Review by kyledurand — APPROVED** [review]
👏

File: src/components/ResourceList/components/FilterControl/FilterControl.scss [hunk 2]
```diff
@@ -7,9 +7,9 @@
   margin: 0;
   display: flex;
   flex-wrap: nowrap;
-  margin-top: -1 * resource-list(header-padding-small);
-  margin-left: -1 * resource-list(header-padding-small);
-  margin-right: -1 * resource-list(header-padding-small);
+  margin-top: calc(-1 * resource-list(header-padding-small));
+  margin-left: calc(-1 * resource-list(header-padding-small));
+  margin-right: calc(-1 * resource-list(header-padding-small));
```
Comment (kyledurand) [inline src/components/ResourceList/components/FilterControl/FilterControl.scss]: Might have to do `calc(resource-list(value) * -1)` to get stylelint to not complain

**Review by lgriffee — APPROVED** [review]
🔥👏🏻❤️

**Review by lgriffee — APPROVED** [review]

**Pull request merged**